File: app/src/main/java/com/lomza/moviesroom/db/Director.java
package com.lomza.moviesroom.db;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.Index;
import androidx.room.PrimaryKey;

/**
 * @author Antonina
 */
@Entity(tableName = "director",
        indices = {@Index(value = "full_name", unique = true)})
public class Director {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "did")
    public int id;

    @ColumnInfo(name = "full_name")
    @NonNull
    public String fullName;

    @Ignore
    public int age;

    public Director(@NonNull String fullName) {
        this.fullName = fullName;
    }
}

File: app/src/main/java/com/lomza/moviesroom/db/Director.kt
package com.lomza.moviesroom.db

import androidx.room.*
import com.lomza.moviesroom.db.Director.Companion.FULL_NAME
import com.lomza.moviesroom.db.Director.Companion.TABLE_NAME

/**
 * @author Antonina
 */
@Entity(
    tableName = TABLE_NAME,
    indices = [Index(value = [FULL_NAME], unique = true)])
data class Director(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "did") var id: Long = 0,
    @ColumnInfo(name = FULL_NAME) var fullName: String,
    @Ignore var age: Int = 0) {

    constructor() : this(0L, "", 0)

    companion object {
        const val TABLE_NAME = "director"
        const val FULL_NAME = "full_name"
    }
}